src/librustc_mir/borrow_check/nll/renumber.rs

```diff
@@ -62,6 +63,18 @@ impl<'a, 'tcx> MutVisitor<'tcx> for NLLVisitor<'a, 'tcx> {
         debug!("visit_ty: ty={:?}", ty);
     }
 
+    fn process_projection_elem(
+        &mut self,
+        elem: Cow<'b, PlaceElem<'tcx>>,
+    ) -> Cow<'b, PlaceElem<'tcx>> {
+        match elem {
+            Cow::Borrowed(PlaceElem::Field(field, ty)) => {
```

Comment:
You should not make decisions depending on the ownedness. The fact that it happened seems makes me wonder if this is a good idea.

you should make the decision solely on the content, so the patterns should just be `PlaceElem::SomeThing` and never about the ownedness. So I guess in this case it would be

```rust
match *elem {
    PlaceElem::Field(field, ty) => Cow::Owned(PlaceElem::Field(field, self.renumber_regions(ty))),
    _ => elem,
}
```

Reply:
Hey, yes, you're right. The root of the problem is that `process_projection_elem` shouldn't be even taking `Cow` as a parameter. I guess I realized about a simpler way to make this work and forgot to change that part. Have just force pushed with a solution.